seamless-immutable.development.js

```diff
@@ -326,20 +326,24 @@
     return result;
   }
 
+  // Creates empty object, using prototype if present
+  function instantiateEmptyObject() {
+    var prototype = Object.getPrototypeOf(this);
+    return prototype !== Object.prototype ? Object.create(prototype) : {};
+  }
+
   // Finalizes an object with immutable methods, freezes it, and returns it.
   function makeImmutableObject(obj) {
     addPropertyTo(obj, "merge", merge);
     addPropertyTo(obj, "without", without);
     addPropertyTo(obj, "asMutable", asMutableObject);
+    addPropertyTo(obj, "instantiateEmptyObject", instantiateEmptyObject);
```

Comment:
Let's inline `instantiateEmptyObject` into `makeImmutableObject` so we can just do the prototype check once instead of every time we instantiate an object:

```javascript
var prototype = Object.getPrototypeOf(this);
var instantiateEmptyObject =
  (prototype === Object.prototype)
    ? instantiateEmptyObject
    : (function() { return Object.create(prototype); });

// ... the other addPropertyTo calls go here ... //

addPropertyTo(obj, "instantiateEmptyObject", instantiateEmptyObject);
```
...and then, elsewhere in the file:
```javascript
function instantiateEmptyObject() { return {}; }
```